src/Runner.Worker/ActionManager.cs

```diff
@@ -795,97 +792,11 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
             string link = downloadInfo?.TarballUrl;
 #endif
 
+            // i will remove the trace.info later, it here for better diff.
             Trace.Info($"Save archive '{link}' into {archiveFile}.");
             try
             {
-                int retryCount = 0;
```

Comment:
i move the entire download logic behind the new `DownloadRepositoryArchive()`
Should not have impact on functionality.